### PR TITLE
Set `bid` when creating job objects, so that `batch` can be used within the jobs

### LINKED ISSUE #44: undefined method `jobs' for nil:NilClass
Hey

Having an issue with batches on a few specific workers.

We have a batch, which enqueues jobs, and then those jobs can add to the batch. This has been working totally fine, but for some reason occasionally this happens.

```
undefined method `jobs' for nil:NilClass
```

That line is `batch.jobs do`

However, as you can see, there is a bid set. (it's in custom, is that right? We don't set that manually).

Screenshot 2020-01-20 at 15 39 44

### PR DESCRIPTION
**Pull request by scottrobertson**
Fixes #44

### CODE REVIEW

File: lib/faktory/processor.rb [hunk 1]
```diff
@@ -133,6 +133,7 @@ def dispatch(payload)
             klass  = constantize(payload['jobtype'.freeze])
             jobinst = klass.new
             jobinst.jid = payload['jid'.freeze]
+            jobinst.bid = payload['custom'.freeze].to_h['bid'.freeze]
```
Comment (scottrobertson) [inline lib/faktory/processor.rb]: @mperham i don't really understand the usage of `.freeze` here, but i just kept it the same as the line above

**Comment by scottrobertson** [comment]
Now that this is fixed, i am getting the following:

```
Faktory::CommandError: Must open batch before modifying it
```

File: lib/faktory/processor.rb (hunk 1 above)
Comment (mperham) [reply lib/faktory/processor.rb]: This is already done by faktory/middleware/batch, which is required by faktory -> faktory/batch.

Comment (scottrobertson) [reply lib/faktory/processor.rb]: Hmm ok. `bid` is 100% not set when accessing from within a job.

Comment (mperham) [reply lib/faktory/processor.rb]: This isn't a good sign:

```
2020-01-20T23:18:05.165Z 65683 TID-owmy4alks DEBUG: Client Middleware: Faktory::Middleware::Batch::Client
2020-01-20T23:18:05.165Z 65683 TID-owmy4alks DEBUG: Worker Middleware: 
```

Comment (scottrobertson) [reply lib/faktory/processor.rb]:
```
Faktory.configure_worker do |config|
  config.client_middleware do |chain|
    chain.add Faktory::Middleware::Batch::Client
  end
  config.worker_middleware do |chain|
    chain.add Faktory::Middleware::Batch::Worker
  end
end
```

Adding some logging in here, and it aint getting there.

Comment (scottrobertson) [reply lib/faktory/processor.rb]: @mperham worth noting, the middlewhere chain code is working, as i built a Honeybadger plugin and it's fine.

```
2020-01-20T23:25:02.610Z 43527 TID-oxvve19e8 DEBUG: Worker Middleware: Honeybadger::Plugins::Faktory::Middleware
```

```
::Faktory.configure_worker do |faktory|
  faktory.worker_middleware do |chain|
    chain.prepend Middleware
  end

  faktory.error_handlers << lambda do |ex, params|
    opts = {parameters: params}
    Honeybadger.notify(ex, opts)
  end
end
```

**Pull request closed without merging**